src/tutorial/errors.md

```diff
@@ -186,10 +186,70 @@ you get this output:
 
 In cases where your code doesn't literally contain the file name,
 it'd be very hard to tell which file was `NotFound`.
+There are multiple ways to deal with this.
 
-<aside class="todo">
+For example, we can create our own error type,
+and then use that to build a custom error message:
 
-**TODO:** Replace `?` with `.with_context(|_| format!("could not read file {}", args.path))`
-[Issue #65](https://github.com/rust-lang-nursery/cli-wg/issues/65)
+```rust
+#[derive(Debug)]
+struct CustomError(String);
 
-</aside>
+fn main() -> Result<(), CustomError> {
+    let path = "test.txt";
+    let content = std::fs::read_to_string(path)
+        .map_err(|err| CustomError(format!("Error reading `{}`: {}", path, err)))?;
+    println!("file content: {}", content);
+    Ok(())
+}
+```
+
+Now,
+running this we'll get our custom error message:
+
+> Error: CustomError("Error reading `test.txt`: No such file or directory (os error 2)")
+
+Not very pretty,
+but we can easily adapt the debug output for our type later on.
+
+This pattern is in fact very common.
+It has one problem, though:
+We don't store the original error,
+only its string representation.
+The often used [`failure`] library has a neat solution for that:
+Similar to our `CustomError` type,
+it has a `Context` type
+that contains a description as well as the original error.
+The library also brings with it an extension trait
+that adds `context()` and `with_context()` methods to `Result`.
```

Comment:
Link to this extension trait here by name rather than just indicating that it exists ... Also, link to the actual docs for `context` and `with_context` and `Context`.